src/frontend/js/hooks/useUnionResource/index.ts

```diff
@@ -90,6 +90,8 @@ const useUnionResource = <
   const eofRef = useRef<Record<string, number>>(queryClient.getQueryData(eofQueryKey) ?? {});
   log('eof', eofRef.current);
 
+  const [unionQueryKey, setUnionQueryKey] = useState<string>();
```

Comment:
I'm not quit sure you need to use a state here.

Instead you could use a useMemo.